src/shared/extensions/redhat-yaml/redhat-yaml.ts

```diff
@@ -0,0 +1,48 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { ExtensionAPI as SchemaApi } from './schema-extension-api'
+import { readFileSync } from 'fs-extra'
+
+const EXTENSION_ID = 'redhat.vscode-yaml'
+const AWS_SCHEME = 'aws-schema'
+const mappings: Map<string, vscode.Uri> = new Map()
+
+async function getSchemaApi(): Promise<SchemaApi> {
+    const ext = vscode.extensions.getExtension(EXTENSION_ID)
+    if (ext !== undefined) {
+        return (await ext.activate())
+    }
+    throw new Error('Redhat YAML extension does not exist')
```

Comment:
Uncaught error. We cannot guarantee that a user has the YAML extension installed and we shouldn't have any adverse effects if they don't have it.

Reply:
True. The extension logic should be paired with a way to determine whether or not the extension is installed.